src/gateway-express.ts

```diff
@@ -14,19 +14,41 @@ function gateway_express(this: any, options: any) {
 
 
   async function handler(req: any, res: any, next: any) {
-    let body = req.body
-    let json = 'string' === typeof (body) ? parseJSON(body) : body
+    const body = req.body
+    const json = 'string' === typeof body ? parseJSON(body) : body
+
     if (json.error$) {
       return res.status(400).send(json)
     }
-    else {
-      let out: any = await gateway(json, { req, res })
-      if (out.done$) {
+
+    try {
+      const out: any = await gateway(json, { req, res })
+
+      if (out?.done$) {
         return next()
       }
-      else {
-        res.send(out)
+
+      if (out?.error$ && !options.bypass_express_error_handler) {
+        // NOTE: Here we are passing the object with information about
+        // the error to the Express' error handler, which allows users
+        // to handle errors in their application.
+        //
+        // This is useful, for example, when you want to return an HTTP
+        // 404 for the 'act_not_found' error; - or handle any other error
+        // that you defined and threw inside a Seneca instance.
+        //
+        return next(out)
       }
+
+      return res.send(out)
+    } catch (err) {
```

Comment:
isn't this redundant? catch and rethrow is the same as no catch

Reply:
@rjrodger I introduced the construct deliberately in order to communicate to the reader that if the `gateway` function call throws, then we do not handle that error. Omitting the construct could erroneously lead the reader into thinking there was a bug.